tests/kafkatest/tests/streams/streams_upgrade_test.py

```diff
@@ -635,10 +635,19 @@ def verify_metadata_no_upgraded_yet(self):
             if len(found) > 0:
                 raise Exception("Kafka Streams failed with 'group member upgraded to metadata 4 too early'")
 
-    def get_topics_count(self):
-        count = 0
+    def confirm_topics_on_all_brokers(self, expected_topic_set):
         for node in self.kafka.nodes:
-            topic_list = self.kafka.list_topics("placeholder", node)
-            for topic in topic_list:
-                count += 1
-        return count
+            match_count = 0
+            # need to iterate over topic_list_generator as kafka.list_topics()
+            # returns a python generator so values are fetched lazily
+            # so we can't just compare directly we must iterate over what's returned
+            topic_list_generator = self.kafka.list_topics("placeholder", node)
+            for topic in topic_list_generator:
+                if topic in expected_topic_set:
```

Comment:
Thinking about this I could add the returned topics from a generator to a set and make an `expected == actual` comparison but I also think this check is sufficient and I don't have to sort the sets before comparison. Also by creating a set of topics found on the broker, I'd have to remove any internal topics as well, further complicating the logic (admittedly not by much).

Reply:
Yeah, I think what you have is fine, since we have to iterate over the generator one way or another, and the set comparison it's really straightforward either.

I guess the only downside of this algorithm is that, if the broker nefariously listed the same topic multiple times, it would give us a false positive. But I seriously doubt that can happen.